src/cryptography/fernet.py

```diff
@@ -134,6 +141,20 @@ def __init__(self, fernets):
     def encrypt(self, msg):
         return self._fernets[0].encrypt(msg)
 
+    def rotate(self, msg):
+        timestamp, data = self._fernets[0]._get_unverified_token_data(msg)
+        for f in self._fernets:
+            try:
+                p = f._decrypt_data(data, timestamp, None)
+                break
+            except InvalidToken:
+                pass
+        else:
+            raise InvalidToken
+
+        iv = os.urandom(16)
```

Comment:
Why does `_encrypt_from_parts` take an `iv`? The only other caller also does this `os.urandom` call. I don't think we want to have two places in our code where we generate the random IV. Maybe we should modify `_encrypt_from_parts` to just take data and timestamp and hoist iv generation into the method?

Reply:
For the tests.

Reply:
Ah. Well that's a good reason. I still don't like os.urandom being called in two places, but maybe I need to just be okay with that.